frontend/src/pages/RunDetails.test.tsx

```diff
@@ -1588,6 +1588,49 @@ describe('RunDetails', () => {
       expect(history.location.pathname).toEqual('/initial-path');
     });
   });
+
+  describe('ReducedGraphSwitch', () => {
+    it('shows a simplified graph', async () => {
+      testRun.pipeline_runtime!.workflow_manifest = JSON.stringify({
+        ...WORKFLOW_TEMPLATE,
+        status: {
+          nodes: {
+            node1: { id: 'node1', children: ['node2', 'node3'] },
+            node2: { id: 'node2', children: ['node3'] },
+            node3: { id: 'node3' },
+          },
+        },
+      });
+      const tree = render(<RunDetails {...generateProps()} />);
+      await getRunSpy;
+      await TestUtils.flushPromises();
+      const snap = `
+      <pre
+        data-testid="graph"
+      >
+        Node node1
+        Node node1-running-placeholder
+        Node node2
+        Node node2-running-placeholder
+        Node node3
+        Node node3-running-placeholder
+        Edge node1 to node1-running-placeholder
+        Edge node2 to node2-running-placeholder
+        Edge node3 to node3-running-placeholder
+        Edge node1 to node2
+        Edge node1 to node3
+        Edge node2 to node3
+      </pre>
+    `;
+      expect(tree.getByTestId('graph')).toMatchInlineSnapshot(snap);
```

Comment:
When we use snapshot testing, jest will auto generate the screenshot in place. So it's not good changing that to a variable.

Two better alternatives, use toEqual instead.
Or use snapshot-diff for the diff part, you should be able to find examples in the code base.

Reply:
@Bobgy I am not sure I follow your argument. I mostly followed the same approach as here: https://github.com/kubeflow/pipelines/blob/cc7cac43b92fdcf6f7672978a9cd1f649fdd815c/frontend/src/pages/RunDetails.test.tsx#L645-L654

Reply:
Sure, let me explain, the string between matchinlinesnapshot is generated by jest. You moved the text to be a variable, that can prevent future regeneration of the inline snapshot, so that's not preferred.

https://jestjs.io/docs/en/snapshot-testing

Reply:
Oh wow! This is super cool, I hadn't realized inline snapshots worked like this. Thanks!